grr/server/grr_response_server/gui/ui/components/timestamp/relative_timestamp_pipe.ts

```diff
@@ -0,0 +1,50 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {DatePipe} from '@angular/common';
+
+@Pipe({
+  name: 'relativeTimestamp'
+})
+export class RelativeTimestampPipe extends
+  DatePipe implements PipeTransform {
+
+  readonly oneDay = 86400000; // One day in millis
+  readonly oneHour = 3600000; // One hour in millis
+  readonly oneMin = 60000; // One minute in millis
+
+  transform(date: Date, absoluteOnly: any): any {
+    if (date === undefined) {
+      return "-";
+    }
+
+    if (absoluteOnly) {
+      return this.getAbsoluteTimestamp(date);
+    }
+
+    var currentDate = new Date();
+    var timeDifference = currentDate.getTime() - date.getTime();
+    var relativeDate = new Date();
+    relativeDate.setTime(timeDifference);
+
+    if (timeDifference < this.oneMin / 2) {
+      return "Just now";
+    }
+
+    if (timeDifference < this.oneHour) {
+      return super.transform(relativeDate, "m'min ago'");
+    }
+
+    if (timeDifference < this.oneDay) {
+      return super.transform(relativeDate, "H'h'm'min ago'");
```

Comment:
`relativeDate` is not actually a time difference, but `timeDifference` added onto the unix epoch. Per default, DatePipe displays any Date in the computer's local timezone. Thus, a timeDifference of 1 hour (`timeDifference = 3600000`) can translate to a `relativeDate` of `1970-01-01 02:00:00 GMT+0100`, which in turn would be displayed as `2h0min ago` to the user.

Reply:
Sorry about this, you're right!